lib/ecto/adapters/postgres/connection.ex

```diff
@@ -453,6 +453,31 @@ if Code.ensure_loaded?(Postgrex) do
     end
 
     defp using_join(%{joins: []}, _kind, _prefix, _sources), do: {[], []}
+
+    defp using_join(%{joins: joins} = query, :update_all, prefix, sources) do
+      {inner_joins, other_joins} = Enum.split_with(joins, & &1.qual == :inner)
+
+      if inner_joins == [] and other_joins != [] do
+        error!(query, "Need at least one inner join to use other joins with update_all")
+      end
```

Comment:
My understanding of the code is that first we need to have all inner joins, right? In that case, we need this:

```suggestion
      {inner_joins, other_joins} = Enum.split_while(joins, & &1.qual == :inner)

      if inner_joins == [] and other_joins != [] do
        error!(query, "Need at least one inner join at the beginning to use other joins with update_all")
      end
```

Otherwise the joins will be out of order?

Reply:
Indeed, a quick test run in `psql` confirms:

```
ecto_sql_pr_457=# UPDATE students student
SET
  name = person.name,
  tutor_name = tutor.name
FROM persons person
LEFT JOIN tutors tutor ON tutor.id = person.tutor_id
WHERE student.person_id = person.id
;
UPDATE 0
ecto_sql_pr_457=# UPDATE students student
SET
  name = person.name,
  tutor_name = tutor.name
LEFT JOIN tutors tutor ON tutor.id = person.tutor_id
FROM persons person
WHERE student.person_id = person.id
;
ERROR:  syntax error at or near "LEFT"
LINE 5: LEFT JOIN tutors tutor ON tutor.id = person.tutor_id
        ^
ecto_sql_pr_457=#
```

...the inner joins, specified via `FROM`, need to come first.

Reply:
Oops, applying this broke the tests since they checked for the error message. ☺️ 

Just fixed that. 🤓